### DISCUSSION TITLE
Feature Requests and suggestions

### DISCUSSION THREAD

**Discussion by vgarleanu** [Dusk-Labs/dim#291] [Ideas]
Hey all, I'm opening this discussion so that people can drop in and vote on feature requests. If you have any feature requests just drop them in below, and if you would like to see the feature implemented, simply vote on it by upvoting.

**Comment by coucoulala** [comment]
External authentication with automatic account provisioning, ideally OIDC otherwise LDAP.

**Comment by Th3Whit3Wolf** [comment]
Upscaler support like [Anime4K](https://github.com/bloc97/Anime4K)

**Comment by evanrichter** [comment]
"Mark as played" in Web UI

**Comment by evanrichter** [comment]
Smarter detection of "watched". Mark movie as watched even if only watched up until the credits.

**Comment by DesignsWithDavid** [comment]
Custom intro that plays before all videos (let the Netflix intro).  Similar to a feature in Jellyfin

**Comment by DesignsWithDavid** [comment]
Pluggable scripts that can modify videos and their metadata while the server is not in use.  For example, LiveTV recordings could run a commercial detection model to later allow for auto-skipping, movies without subtitles could run a subtitle generating model, movies of lower quality could run a model that increases the resolution, or transcodes could be done that create duplicates of commonly watched videos in formats that are already ready for streaming to the most commonly used devices.  There could even be models for detecting inappropriate content in the video and audio, generating lists of items and people in the videos for improving search results, or determining "similar" videos for recommendation purposes.  All of these things could queue up as movies/shows are added to the database and they could run at set times of day, or whenever no one is streaming video.

tl;dr - A means of running machine learning models on videos while the server is not in high use.

**Comment by merlin699** [comment]
Possibility the select another language than english in the frontend, including IMDb.

**Comment by jeremytodd1** [comment]
I'd like to have the "Skip Intro" feature that Plex Pass brings. Being able to easily skip a show's intro is great.

**Comment by dvmrry** [comment]
I know this is not a simple ask, but the ability to have transcoding distributed across multiple nodes would be incredible.

**Reply by vgarleanu** [reply]
This has been discussed a lot and will come as part of a paid plugin in the future alongside more features that are directly related to scaling, such as multi-gpu support, load-balancing, etc.

**Reply by DesignsWithDavid** [reply]
@vgarleanu 
Would you be able to expand on your paid plugin discussions?  Are those discussions public?  I am very interested in hearing what DIM is planning with respect to financing the project while maintaining things as open-source as possible

**Reply by vgarleanu** [reply]
Sure, I should create a separate discussion for this, maybe even open source the business plan.

> Are those discussions public?

Somewhat. Most of the discussions are just me brainstorming in the DMs with my friends and other contributors, but some conversations have been had in our Discord server (which you can btw join here: https://discord.gg/gBPyQ7NVah).

> Would you be able to expand on your paid plugin discussions?

The core funding model of dim will come as form of a paid subscription to a set of plugins that extend the functionality of dim beyond the basics. 

One of the most important things we try to do here is improve the UX as much as possible. One thing that I've realised, from user feedback is that a lot of people don't exactly like port-forwarding, maintaining firewall rules, etc, as such we think it would be a good idea to create some sort of dim relay, which would improve the accessibility and discover-ability of dim instances.

Another feature that I find is quite niche but could see a market is adding better scaleability to dim in several forms. The main one that excited me is multi-GPU transcode load-balancing as well as remote transcoding. These features are niche enough that I think not too many users would be upset if we make it a paid feature. Another one that I've thought about is federated scaling, ie you run multiple dim instances at home and they inter-connect with each other.

In general we will be targeting a lot features that sys-admins kind of like. Grafana integration and metrics/activity page could be paid (although I'm not too sure on this just yet). Fine-grained permission control and sharing restrictions, and more advanced server settings as well. Hosted instances of dim would also be interesting but I'm not sure how much demand there would be for this, and plus the market is already saturated pretty well (seedboxes, etc).

The main reason dim requires monetization is that developing as many clients as we need to stay viable is expensive, both time and resources wise. At the moment I am sponsoring this project myself, and I think I could manage sponsoring the development of a mobile app and maybe a desktop app, but anything beyond that is out of scope without a clear strategy for funding.

Another thing that I've been thinking about is having more exotic clients be paid/require a subscription. This will undoubtedly piss off a lot of people, but if it gains some velocity, it can help us deliver a better experience than a half-assed client that was put together within a week and is maintained little simply because of lack of manpower and time.

I have some other ideas but I haven't thought about them long enough to figure out how hard it'd be to manifest. Some are pretty grandiose but might just not work, so I'd rather not list them just yet.

If you have any suggestions, or if you have any concerns, I'd be interested in hearing them.

p.s. I keep seeing people call dim "DIM", which I think might confuse others, thinking it means Destiny Item Manager. I wonder how this started :)

**Comment by Yanom1212** [comment]
I can't wait for this to evolve further. 
Some idea:
Plugins for TMDb, TheTVDb, etc.
Metadata Language. 
Advanced Transcode setting.

**Comment by ironduke151** [comment]
Like work have done so far.
one thing can add is choice of scrapper to work with movies or tv shows

**Comment by jonwilliams84** [comment]
Would be nice to be able to add remote transcode workers and maybe a kubernetes operator to create transcode pods on demand. This would allow you to run the front-end on any hardware and orchestrate the heavy lifting when it's required. It also makes it scalable which is something missing from the other kids on the block. With this and Android/iOS/Tizen/WebOs clients I would move in a heartbeat.

**Comment by soupglasses** [comment]
Having multiple accounts be able to join a watch-party. Related to #226

**Comment by RenMu-V** [comment]
想问下大家是否需要arm64平台的安装支持呢？
I would like to ask if you need support for the arm64 platform installation?

**Comment by ghost** [comment]
Streaming with a torrent plugin like stremio.

**Comment by sharkAndshark** [comment]
better if i can search and add subtitles in dim

**Reply by enigmaoftech** [reply]
The ability to set subtitle preferences around forced or on for subtitles too.

**Comment by enigmaoftech** [comment]
Library Sort & Filters

Alphabet on the side to quickly jump to places in larger libraries
Sort by
- Title
- Year
- Release Date
- Rating
- Content Rating
- Duration
- Resolution
- Bitrate
- Random “This will be useful with collections”

**Comment by enigmaoftech** [comment]
Collections
The ability to create cross library collections and add them to the home screen too.
Say collections based on trending and holiday collections you could stick on the home screen.
Other collections would just be in a collections grouping. 

Take a look at plex meta manager and all the great feature they have brought in. Being able to api in and do these same things or end up building directly in would be amazing.

**Comment by enigmaoftech** [comment]
Multiple Ratings
	IMDB – Critic Rating
	Rotten Tomatoes – Audience Rating
	TMDB – User Rating

**Comment by enigmaoftech** [comment]
User Management
Grant/Deny permissions to the libraries also based on folder underneath
Movie Library
	- Movies
	- Movies 4k
TV Shows
	- TV
	- TV 4k
	- TV Kids

Parental Controls
	Control based on rating
	Control based on tag

**Comment by enigmaoftech** [comment]
Library versioning support for picking 1080/4k/extended edition etc

**Comment by enigmaoftech** [comment]
Live TV support with M3U

**Comment by enigmaoftech** [comment]
If/when you support mobile devices, I would suggest trying to stick with the main ones.

- iOS
- Android
- NVIDIA Shield
- Roku
- Apple TV

I did see that you had some sample code out there for the frame of iOS and Android. Are they ready for testing?

**Comment by enigmaoftech** [comment]
Progress bars for libraries. That way you know what is going on.

**Comment by enigmaoftech** [comment]
Activity dashboard on users to see all users actively playing media, progress, transcoding.
Also add the ability to see and track activity like tautulli or varys.

**Comment by enigmaoftech** [comment]
On Home Screen have Recently Released. This will organize movies based on release date and not added at date. Keep the Recently added as another section as well.

I like having on the Home Screen
Continue Watching
*Holiday Collection Timed based
Recently Released Movies 
Trending Movies 
Recently Released Shows
Trending Shows
